### PR TITLE
Release v1.5.6

### PR DESCRIPTION
**Pull request by yakov-g**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
- add error message when trying to run perf-test for unstructured mode
- RAPTOR-5328: set HOME to /opt and WORKDIR to /opt/code for every env
- add a memo regarding DRUM usage in Windows
- split julia env image into base image and env itself

### CODE REVIEW

File: custom_model_runner/README.md [hunk 1]
```diff
@@ -8,6 +8,8 @@ DRUM can also:
 - perform model validation tests, e.g., checking model functionality on corner cases, like null values imputation.
 - run models in a Docker container.
 
+***DataRobot DRUM is only tested to support Linux/macOS operating systems. To run DRUM in WIndows 10 please use WSL (Windows Subsystem for Linux).***
```
Comment (zohar-mizrahi) [inline custom_model_runner/README.md:11]: nit: `Windows 10` (capital `I`)

File: custom_model_runner/datarobot_drum/drum/args_parser.py [hunk 2]
```diff
@@ -920,6 +920,12 @@ def verify_options(options):
                 CMRunnerArgsRegistry._parsers[ArgumentsOptions.NEW].print_help()
                 exit(1)
         elif options.subparser_name in [ArgumentsOptions.SERVER, ArgumentsOptions.PERF_TEST]:
+            if (
+                options.subparser_name == ArgumentsOptions.PERF_TEST
+                and options.target_type == TargetType.UNSTRUCTURED.value
+            ):
+                print("Performance testing is not implemented for the unstructured models.")
```
Comment (zohar-mizrahi) [inline custom_model_runner/datarobot_drum/drum/args_parser.py:927]: nit: I'd remove the word `the`, as it's a kind of general message

File: public_dropin_environments/julia_mlj/Dockerfile [hunk 3]
```diff
@@ -36,10 +35,11 @@ RUN julia "$HOME/julia/sysim.jl" && \
     chmod -R 777 $HOME/.julia/logs/manifest_usage.toml && \
     rm julia-1.5.4-linux-x86_64.tar.gz
 
-ENV CODE_DIR=/opt/code 
-ENV ADDRESS=0.0.0.0:8080 WITH_ERROR_SERVER=1
+ENV HOME=/opt CODE_DIR=/opt/code ADDRESS=0.0.0.0:8080
+WORKDIR /opt/code
```
Comment (zohar-mizrahi) [inline public_dropin_environments/julia_mlj/Dockerfile:39]:
```suggestion
WORKDIR ${CODE_DIR}
```

File: public_dropin_environments/java_codegen/Dockerfile [hunk 4]
```diff
@@ -14,7 +14,9 @@ RUN pip3 install -U --upgrade-strategy eager --no-cache-dir -r dr_requirements.t
 # Code from the custom model tarball can overwrite the code here
 COPY ./ /opt/code/
 
-ENV CODE_DIR=/opt/code ADDRESS=0.0.0.0:8080
+
+ENV HOME=/opt CODE_DIR=/opt/code ADDRESS=0.0.0.0:8080
+WORKDIR /opt/code
```
Comment (zohar-mizrahi) [inline public_dropin_environments/java_codegen/Dockerfile:19]:
```suggestion
WORKDIR ${CODE_DIR}
```

File: public_dropin_environments/julia_mlj/Dockerfile [hunk 5]
```diff
@@ -36,10 +35,11 @@ RUN julia "$HOME/julia/sysim.jl" && \
...
 
+ENV WITH_ERROR_SERVER=1
 #Uncomment the following line to switch from Flask to uwsgi server
-ENV PRODUCTION=1 MAX_WORKERS=1 SHOW_STACKTRACE=1
+#ENV PRODUCTION=1 MAX_WORKERS=1 SHOW_STACKTRACE=1
 
 ENTRYPOINT ["/opt/code/start_server.sh"]
```
Comment (zohar-mizrahi) [inline public_dropin_environments/julia_mlj/Dockerfile:45]:
```suggestion
ENTRYPOINT ["${CODE_DIR}/start_server.sh"]  
```

File: public_dropin_environments/python3_keras/Dockerfile [hunk 6]
```diff
@@ -20,7 +20,8 @@ RUN pip3 install -r requirements.txt --no-cache-dir && \
 # Code from the custom model tarball can overwrite the code here
 COPY ./ /opt/code/
 
-ENV CODE_DIR=/opt/code ADDRESS=0.0.0.0:8080
+ENV HOME=/opt CODE_DIR=/opt/code ADDRESS=0.0.0.0:8080
+WORKDIR /opt/code
```
Comment (zohar-mizrahi) [inline public_dropin_environments/python3_keras/Dockerfile:24]:
```suggestion
WORKDIR ${CODE_DIR}
```

File: public_dropin_environments/python3_pmml/Dockerfile [hunk 7]
```diff
@@ -22,7 +22,8 @@ RUN pip3 install -r requirements.txt --no-cache-dir && \
 # Code from the custom model tarball can overwrite the code here
 COPY ./ /opt/code/ 
 
-ENV CODE_DIR=/opt/code ADDRESS=0.0.0.0:8080
+ENV HOME=/opt CODE_DIR=/opt/code ADDRESS=0.0.0.0:8080
+WORKDIR /opt/code
```
Comment (zohar-mizrahi) [inline public_dropin_environments/python3_pmml/Dockerfile:26]:
```suggestion
WORKDIR ${CODE_DIR}
```

File: public_dropin_environments/python3_pytorch/Dockerfile [hunk 8]
```diff
@@ -20,7 +20,8 @@ RUN pip3 install -r requirements.txt --no-cache-dir && \
 # Code from the custom model tarball can overwrite the code here
 COPY ./ /opt/code/
 
-ENV CODE_DIR=/opt/code ADDRESS=0.0.0.0:8080
+ENV HOME=/opt CODE_DIR=/opt/code ADDRESS=0.0.0.0:8080
+WORKDIR /opt/code
```
Comment (zohar-mizrahi) [inline public_dropin_environments/python3_pytorch/Dockerfile:24]:
```suggestion
WORKDIR ${CODE_DIR}
```

File: public_dropin_environments/python3_xgboost/Dockerfile [hunk 9]
```diff
@@ -20,7 +20,8 @@ RUN pip3 install -r requirements.txt --no-cache-dir && \
 # Code from the custom model tarball can overwrite the code here
 COPY ./ /opt/code/
 
-ENV CODE_DIR=/opt/code ADDRESS=0.0.0.0:8080
+ENV HOME=/opt CODE_DIR=/opt/code ADDRESS=0.0.0.0:8080
+WORKDIR /opt/code
```
Comment (zohar-mizrahi) [inline public_dropin_environments/python3_xgboost/Dockerfile:24]:
```suggestion
WORKDIR ${CODE_DIR}
```

File: public_dropin_environments/r_lang/Dockerfile [hunk 10]
```diff
@@ -13,7 +13,8 @@ RUN pip3 install -U --upgrade-strategy eager -r dr_requirements.txt --no-cache-d
 
 COPY ./ /opt/code/
 
-ENV CODE_DIR=/opt/code ADDRESS=0.0.0.0:8080
+ENV HOME=/opt CODE_DIR=/opt/code ADDRESS=0.0.0.0:8080
+WORKDIR /opt/code
```
Comment (zohar-mizrahi) [inline public_dropin_environments/r_lang/Dockerfile:17]:
```suggestion
WORKDIR ${CODE_DIR}
```

**Review by zohar-mizrahi — DISMISSED** [review]

**Comment by yakov-g** [comment]
jarvis test cmrunner_integration

**Review by meltzg — CHANGES_REQUESTED** [review]
what are all these new files related to julia? were they meant to be added in this PR?

**Comment by yakov-g** [comment]
jarvis test functional_inference_model_templates

**Comment by yakov-g** [comment]
> what are all these new files related to julia? were they meant to be added in this PR?

We already have julia env in the repo, but it takes more than timeout to build this env during the test. So I had to split it into a base image and public env image itself.
Just removed files i moved to the base image from the public env.

**Review by timsetsfire — APPROVED** [review]
looks great!

**Review by meltzg — APPROVED** [review]
